tests/cases/compiler/NonNullableInNonStrictMode.ts

```diff
@@ -0,0 +1,18 @@
+// @strict: false
+
+// These should all resolve to never
+
+type T0 = NonNullable<null>;
+type T1 = NonNullable<undefined>;
+type T2 = null & {};
+type T3 = undefined & {};
+type T4 = null & undefined;
+type T6 = null & { a: string } & {};
+
+// Repro from #50519
+
+type NonNullableNew<T> = T & {};
+type NonNullableOld<T> = T extends null | undefined ? never : T;
+
+type IsNullWithoutStrictNullChecks = NonNullableNew<null>;

```

Comment:
nit: this name seems to be confusing to me since it doesn't actually describe the expected result (if I understand correctly) but rather it refers to what was a bug